downloads/swan-lake-release-notes/swan-lake-beta4/RELEASE_NOTE.md

```diff
@@ -0,0 +1,821 @@
+
+    if res is error {
+        return error("Invalid Value", res);
+    }
+
+    // This was previously allowed but now a compilation error since `res`'s type is now `int` and doesn't include `error`.
+    int intRes = check res;
+
+    arr.push(intRes);
+}
+```
+
+##### Restrictions on Assignments to Narrowed Variables within Loops
+
+Stopped the possibility to assign a value to a variable that was narrowed outside the statement within a `while` statement or a `foreach` statement. Unless the loop terminates after the assignment (i.e., at the end of the loop body and at every `continue` statement), there must be no possibility that a narrowed variable to be assigned.
+
+The example below which previously resulted in a runtime panic will now result in a compilation error.
+
+```ballerina
+function validate(int?[] arr) returns boolean {
+    int? value = let int length = arr.length() in length > 0 ? length : ();
+
+    if value is int {
+        foreach int? item in arr {
+            int currentValue = value;
+
+            if item is () {
+                value = (); // Error: invalid attempt to assign a value to a variable narrowed outside the loop.
+                continue;
+            }
+
+            return item < value;
+        }
+    }
+
+    return false;
+}
+ 
+public function main() {
+   boolean validationRes = validate([(), 2, 1]);
+}
+```
+
+##### Change in Expected Return Statements in a Function with an Optional Type as the Return Type
+
+Updated a function having an optional type that is not a subtype of `error?` as the return type to explicitly return a value. A warning is emitted when such a function does not explicitly return a value and falls off at the end of the function body. 
+
+```ballerina
+function parse(string str) returns int? { // Now results in a warning. 
+    int|error a = int:fromString(str);
+    if a is int {
+        return a;
+    }
+}
+```
+
+#### Bug Fixes and Breaking Changes
+
+- Disallowed the trailing dot format of the floating-point literal to avoid lexical ambiguity.
+
+    ```ballerina
+    // The below are now disallowed.
+    decimal d1 = 2.;
+    decimal d2 = 2.d;
+    decimal d3 = 2.D;
+    decimal d4 = 2.e12;
+    float f1 = 2.f;
+    float f2 = 2.F;
+    float f3 = 0x1A.;
+    float f4 = 0x1A.p4;
+
+    // The below can be used instead.
+    decimal d11 = 2.0;
+    decimal d12 = 2.0d;
+    decimal d13 = 2.0D;
+    decimal d14 = 2.0e12;
+    float f11 = 2.0f;
+    float f12 = 2.0F;
+    float f13 = 0x1A.0;
+    float f14 = 0x1A.0p4;
+    ```
+
+- Disallowed intervening white spaces in the qualified identifier to avoid a parsing ambiguity between the ternary conditional expression and qualified identifier.
+  
+    ```ballerina
+    import ballerina/io;
+
+    public function main() {
+        io:print("Ballerina"); // Valid.
+        io : print("Ballerina"); // compilation error: intervening whitespaces are not allowed in a qualified identifier.
+    }
+```
```

Comment:
It seems due to above incorrect indent in backticks all the following text has become a code.

Reply:
Fixed.